Assets/Scripts/Networking/TesterMenu.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.UI;
+using Hashtable = ExitGames.Client.Photon.Hashtable;
+
+
+public class TesterMenu : MonoBehaviour
+{
+    public ConnectionHandler ConnectionHandler;
+
+    private float startTime = 1;
+
+    public Text StateText;
+
+	// Use this for initialization
+	void Start ()
+	{
+	    //_connectionHandler = Object.FindObjectOfType<ConnectionHandler>();
+	    startTime = Time.time;
+	    var roomProperties = PhotonNetwork.room.CustomProperties;
+	    if (roomProperties.ContainsKey(PhotonPropId.MatchState))
+	    {
+	        var matchState = (MatchState)roomProperties[PhotonPropId.MatchState];
+	        OnMatchState(matchState);
+
+	    }
+    }
+	
+	// Update is called once per frame
+	void Update () {
+	    var roomProperties = PhotonNetwork.room.CustomProperties;
+	    MatchState matchState = MatchState.Wait;
+	    if (roomProperties.ContainsKey(PhotonPropId.MatchState))
+	    {
+	        matchState = (MatchState)roomProperties[PhotonPropId.MatchState];
+
+	    }
+
+        //cycling through some states
+        switch (matchState)
+	    {
+	        case MatchState.Start:
+	            StartingState();
+                break;
+            case MatchState.Play:
+                PlayingState();
+                break;
+	    }
```

Comment:
What about MatchState.Wait?

also SPESS

Reply:
This will throw a failure if we do have `MatchState.Wait` when it hits this switch

Reply:
It shouldn't do anything if it does not have a switch state. Same with `MatchState.End`. No need for `default:` either because we can just ignore the states until we are ready to do something with them